rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +607,140 @@ def _query_lending_balances(
 
         return balances
 
+    def _query_lending_interests_history(
+            self,
+            cursor: 'DBCursor',
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> None:
+        """Queries Binance lending interest history, transforms it
+        into `HistoryBaseEntry` objects and save it in the database.
+        May raise:
+        - RemoteError
+        """
+        ranges = DBQueryRanges(self.db)
+        range_query_name = f'{self.location}_history_events_{self.name}'
+        ranges_to_query = ranges.get_location_query_ranges(
+            cursor=cursor,
+            location_string=range_query_name,
+            start_ts=start_ts,
+            end_ts=end_ts,
+        )
+        for query_start_ts, query_end_ts in ranges_to_query:
+            events = []
+            for lending_type in ('DAILY', 'ACTIVITY', 'CUSTOMIZED_FIXED'):
+                response = self._api_query_list_within_time_delta(
+                    api_type='sapi',
+                    start_ts=query_start_ts,
+                    time_delta=BINANCE_LENDING_INTEREST_HISTORY_INTERVAL_TS,
+                    end_ts=query_end_ts,
+                    method='lending/union/interestHistory',
+                    additional_options={'lendingType': lending_type, 'size': 100},
+                )
+                for idx, entry in enumerate(response):
+                    try:
+                        interest_received = deserialize_asset_amount_force_positive(entry['interest'])  # noqa: E501
+                        if interest_received == ZERO:
+                            continue
+
+                        event = HistoryBaseEntry(
+                            event_identifier=HistoryBaseEntry.deserialize_event_identifier(str(uuid4())),  # noqa: E501
```

Comment:
I would move the event creation out of the try where you have all the try/except just because it makes clearer and easier to read what is happening and what failed. If any of this deserialization fails it will point to this first line instead of the line that failed I believe